### DISCUSSION TITLE
Extending .cshtml view locations

### DISCUSSION THREAD

**Discussion by craftyweb** [OrchardCMS/OrchardCore#13883] [Q&A] [answered]
Hello,

I'm building a large Webshop module in OrchardCore. Because I have a lot of Parts and Fields I have a lot of views inside my Views folder.
I was wondering if I could do somethings to create subfolders for each feature inside of my module and place my views related to that feature inside a views folder that is part of the subfolder.

Here is an example of the structure I would like to create.

![image](https://github.com/OrchardCMS/OrchardCore/assets/61787638/700367b9-4663-4d95-b90f-33e5c7566fac)

After some research I've tried to do something with an IViewLocationExpanderProvider but I can't figure it out how to do so.
Is this the right way or do I need to do something else?

Thanks!

**Comment by jtkech** [comment, marked as answer]
Can be done by implementing a custom `IShapeTemplateHarvester`, you can copy paste the current implementation but for  `SubPaths()` return the view paths you need, then in your module startup register this custom implementation as a scoped service, not to replace the current implementation but as an additional harvester.

**Reply by craftyweb** [reply]
Thanks! Works like a charm!

**Comment by ghost** [comment]
@craftyweb I run into your post while searching for an ecommerce solution on OrchardCore. Are you going to offer yours as a commercial module? Do you have a demo site?

**Reply by hishamco** [reply]
@AngelaGreen0 this is unrelated, but you can have a look at https://github.com/OrchardCMS/OrchardCore.Commerce
